src/lib.rs

```diff
@@ -26,8 +43,16 @@ pub struct Id<'id> {
 }
 
 impl<'id> Id<'id> {
-    // Do not use this function; use the `make_guard!` macro instead.
-    #[doc(hidden)]
+    /// Construct an `Id` with an unbound lifetime.
+    ///
+    /// You should not need to use this function; use [`make_guard!`] instead.
+    ///
+    /// # Safety
+    ///
+    /// This creates an unbound invariant lifetime that people are allowed to
+    /// assume means it was derived from their generative brand. This is the
+    /// "I know what I'm doing" button; restrict the lifetime to a known brand
+    /// immediately to avoid accidentally introducing unsoundness potential.
```

Comment:
To make the safe usage requirements clearer, I would use wording similar to this:

> `Id` holds an invariant lifetime that must be derived from a generative brand. Using this function directly is the "I know what I'm doing" button. Restrict the lifetime to a known brand immediately to avoid introducing unsoundness.